### PR TITLE
Upgrading vendored AWS SDK release to v1.8.19

### PR DESCRIPTION
**Pull request by petderek**
### Summary
Update the vendored AWS SDK

### Implementation details
* Updated dependencies using godep. (See Godeps.json)
* Regenerated client code

### Testing

- [x] Builds on Linux (`make release`)
- [x] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [x] Unit tests on Linux (`make test`) pass
- [x] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [x] Integration tests on Linux (`make run-integ-tests`) pass
- [x] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [x] Functional tests on Linux (`make run-functional-tests`) pass
- [x] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes: no

### Description for the changelog
Update AWS SDK release v1.8.19

### Licensing
This contribution is under the terms of the Apache 2.0 License:  yes

### CODE REVIEW

File: agent/gogenerate/awssdk.go [hunk 1]
```diff
@@ -39,7 +41,7 @@ var typesOnlyTplAPI = template.Must(template.New("api").Parse(`
 {{ end }}
 `))
 
-const copyrightHeader = `// Copyright 2014-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+const copyrightHeader = `// Copyright 2014-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (aaithal) [inline agent/gogenerate/awssdk.go]: minor: Can you modify this to get the `year` string in code so that we don't have to update this every year?

**Comment by aaithal** [comment]
@petderek can you also verify that on a long running Agent built from this code, metrics continue to show up in the ECS console?

File: agent/gogenerate/awssdk.go [hunk 2]
```diff
@@ -41,7 +42,7 @@ var typesOnlyTplAPI = template.Must(template.New("api").Parse(`
 {{ end }}
 `))
 
-const copyrightHeader = `// Copyright 2014-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+const copyrightHeaderTemplate = `// Copyright 2014-%v Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (aaithal) [inline agent/gogenerate/awssdk.go]: Can you changes to `%d` instead of `%v` so that it's more specific

**Comment by petderek** [comment]
@aaithal I verified metrics are working after letting the agent run overnight and launching some tasks just now. Did you have any other concerns with metrics?

**Comment by aaithal** [comment]
@petderek thanks. Nope, just wanted to verify that they show up.

File: agent/ecr/model/generate.go [hunk 3]
```diff
@@ -13,4 +13,4 @@
 
 package model
 
-//go:generate go run ../../gogenerate/awssdk.go -typesOnly=false
```
Comment (aaithal) [inline agent/ecr/model/generate.go]: Who's using the `codegen` tag?

Comment (petderek) [inline agent/ecr/model/generate.go]: The SDK team started using the codegen tag in their generators, so we need this in order to run the generate commands. I put a `+build codegen` in `awssdk.go` as well.

Comment (aaithal) [inline agent/ecr/model/generate.go]: Got it. In the spirit of "self documenting code", can you either modify the commit message or add a comment to state the same?

**Pull request merged**